File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        ArrayList<Integer> list = new ArrayList<>();
        int n1 = nums1.length;
        int n2 = nums2.length;

        int i = 0;
        int j = 0;

        while(i < n1 && j < n2) {
            if(nums1[i] <= nums2[j]) {
                list.add(nums1[i]);
                i++;
            }
            else {
                list.add(nums2[j]);
                j++;
            }
        }

        while(i < n1) {
            list.add(nums1[i]);
            i++;
        }

        while(j < n2) {
            list.add(nums2[j]);
            j++;
        }

        int n = list.size();

        if(n%2 != 0) return list.get(n/2)*1.0 ;
        else return(list.get(n/2 -1) + list.get(n/2)) / 2.0 ;
    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();

        vector<int> merArr;
        int i=0, j=0;

        while(i<n && j<m) {
            if(nums1[i] < nums2[j]) {
                merArr.push_back(nums1[i]);
                i++;
            } else {
                merArr.push_back(nums2[j]);
                j++;
            }
        }

        while(i<n) {
            merArr.push_back(nums1[i]);
            i++;
        }

        while(j<m) {
            merArr.push_back(nums2[j]);
            j++;
        }

        int size = merArr.size();
        double median;
        if(size%2 == 0) {
            median = (merArr[size/2] + merArr[(size-1)/2])/2.0 ;
        } else {
            median =  merArr[(size-1)/2];
        }

        return median;
    }
};